### PR TITLE
Remove Azure Active Directory Authentication Library (ADAL)

### PR DESCRIPTION
**Pull request by JohnSchmeichel**
This removes Azure Active Directory Authentication Library (ADAL) support from the credential provider and all associated environment variables.

### CODE REVIEW

File: CredentialProvider.Microsoft.VSIX/Microsoft.CredentialProvider.swr [hunk 1]
```diff
@@ -19,7 +19,6 @@ folder InstallDir:\Common7\IDE\CommonExtensions\Microsoft\NuGet\Plugins\Credenti
   file source=$(PluginBinPath)\Microsoft.Identity.Client.Extensions.Msal.dll
   file source=$(PluginBinPath)\Microsoft.Identity.Client.NativeInterop.dll
   file source=$(PluginBinPath)\Microsoft.IdentityModel.Abstractions.dll
```
Comment (JohnSchmeichel) [inline CredentialProvider.Microsoft.VSIX/Microsoft.CredentialProvider.swr:21]: >Microsoft.IdentityModel.Abstractions.dll [](http://example.com/codeflow?start=31&length=40)

Unfortunately this one is still used by MSAL

File: CredentialProvider.Microsoft/Util/EnvUtil.cs [hunk 2]
```diff
@@ -21,20 +21,16 @@ public static class EnvUtil
         public const string WindowsIntegratedAuthenticationEnvVar = "NUGET_CREDENTIALPROVIDER_WINDOWSINTEGRATEDAUTHENTICATION_ENABLED";
         public const string ForceCanShowDialogEnvVar = "NUGET_CREDENTIALPROVIDER_FORCE_CANSHOWDIALOG_TO";
 
-        public const string AuthorityEnvVar = "NUGET_CREDENTIALPROVIDER_ADAL_AUTHORITY";
-        public const string AdalFileCacheEnvVar = "NUGET_CREDENTIALPROVIDER_ADAL_FILECACHE_ENABLED";
-        public const string PpeHostsEnvVar = "NUGET_CREDENTIALPROVIDER_ADAL_PPEHOSTS";
```
Comment (JohnSchmeichel) [inline CredentialProvider.Microsoft/Util/EnvUtil.cs:26]: >NUGET_CREDENTIALPROVIDER_ADAL_PPEHOSTS [](http://example.com/codeflow?start=46&length=38)

This one had nothing to do with ADAL, so renaming it. Technically a back-compat change, but considering it's for PPE the existing usage should be low

File: CredentialProvider.Microsoft.Tests/CredentialProviders/Vsts/AuthUtilTests.cs [hunk 3]
```diff
@@ -48,29 +47,26 @@ public void TestCleanup()
         [TestMethod]
         public async Task GetAadAuthorityUri_WithoutAuthenticateHeaders_ReturnsCorrectAuthority()
```
Comment (JohnSchmeichel) [inline CredentialProvider.Microsoft.Tests/CredentialProviders/Vsts/AuthUtilTests.cs:48]: >GetAadAuthorityUri_WithoutAuthenticateHeaders_ReturnsCorrectAuthority [](http://example.com/codeflow?start=26&length=69)

checking if there were already MSAL versions of these and these are now duplicates and can be removed

File: README.md [hunk 4]
```diff
@@ -261,14 +248,11 @@ Cache Location
     deleted, the credential provider will re-create them but any credentials
     will need to be provided again.
 
-    ADAL Token Cache
-    C:\Users\someuser\AppData\Local\MicrosoftCredentialProvider\ADALTokenCache.dat
+    MSAL Token Cache
+    C:\Users\Schmeichel\AppData\Local\.IdentityService
```
Comment (JohnSchmeichel) [inline README.md:252]: >Schmeichel [](http://example.com/codeflow?start=13&length=10)

fix here and below

**Comment by JohnSchmeichel** [comment]
> set NUGET_CREDENTIALPROVIDER_MSAL_ENABLED=true

remove this usage

---
Refers to: test.bat:13 in 07d6c38. [](commit_id = 07d6c38a3d6316a88772d51caeb1eadfbbfa2a21, deletion_comment = False)

**Comment by JohnSchmeichel** [comment]
> set NUGET_CREDENTIALPROVIDER_MSAL_ENABLED=true

"Testing MSAL with broker"
Testing "netcoreapp3.1" with NUGET_CREDENTIALPROVIDER_MSAL_ALLOW_BROKER=true
dotnet run --no-restore --no-build -f "netcoreapp3.1" --project CredentialProvider.Microsoft\CredentialProvider.Microsoft.csproj -- -C -U [https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json](https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json) -V Debug
Testing "net461" with NUGET_CREDENTIALPROVIDER_MSAL_ALLOW_BROKER=true
dotnet run --no-restore --no-build -f "net461" --project CredentialProvider.Microsoft\CredentialProvider.Microsoft.csproj -- -C -U [https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json](https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json) -V Debug
Testing "net6.0" with NUGET_CREDENTIALPROVIDER_MSAL_ALLOW_BROKER=true
dotnet run --no-restore --no-build -f "net6.0" --project CredentialProvider.Microsoft\CredentialProvider.Microsoft.csproj -- -C -U [https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json](https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json) -V Debug
"Testing MSAL without broker"
Testing "netcoreapp3.1" with NUGET_CREDENTIALPROVIDER_MSAL_ALLOW_BROKER=false
dotnet run --no-restore --no-build -f "netcoreapp3.1" --project CredentialProvider.Microsoft\CredentialProvider.Microsoft.csproj -- -C -U [https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json](https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json) -V Debug
Testing "net461" with NUGET_CREDENTIALPROVIDER_MSAL_ALLOW_BROKER=false
dotnet run --no-restore --no-build -f "net461" --project CredentialProvider.Microsoft\CredentialProvider.Microsoft.csproj -- -C -U [https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json](https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json) -V Debug
Testing "net6.0" with NUGET_CREDENTIALPROVIDER_MSAL_ALLOW_BROKER=false
dotnet run --no-restore --no-build -f "net6.0" --project CredentialProvider.Microsoft\CredentialProvider.Microsoft.csproj -- -C -U [https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json](https://pkgs.dev.azure.com/mseng/PipelineTools/_packaging/artifacts-credprovider/nuget/v3/index.json) -V Debug
"All tests passed"

---
In reply to: [1539237895](https://github.com/microsoft/artifacts-credprovider/pull/411#issuecomment-1539237895) [](http://example.com/codeflow?ancestors=1539237895)

---
Refers to: test.bat:13 in 07d6c38. [](commit_id = 07d6c38a3d6316a88772d51caeb1eadfbbfa2a21, deletion_comment = False)

**Review by johnterickson — APPROVED** [review]

**Review by embetten — APPROVED** [review]

File: CredentialProvider.Microsoft/Util/HttpClientFactory.cs [hunk 5]
```diff
@@ -5,11 +5,10 @@
 using System;
 using System.Net.Http;
 using Microsoft.Artifacts.Authentication;
-using IAdalHttpClientFactory = Microsoft.IdentityModel.Clients.ActiveDirectory.IHttpClientFactory;
 
 namespace NuGetCredentialProvider.Util
 {
-    public class HttpClientFactory : MsalHttpClientFactory, IAdalHttpClientFactory
```
[the remark this answers is not in the document]
Comment (JohnSchmeichel) [reply CredentialProvider.Microsoft/Util/HttpClientFactory.cs:11]: they shared the same instance method

**Pull request merged**